backend/schema/validate.go

```diff
@@ -547,24 +548,53 @@ func validateVerbMetadata(scopes Scopes, module *Module, n *Verb) (merr []error)
 
 		switch md := md.(type) {
 		case *MetadataIngress:
-			reqBodyType, reqBody, errs := validateIngressRequestOrResponse(scopes, module, n, "request", n.Request)
+			reqInfo, errs := validateIngressRequest(scopes, module, n, "request", n.Request, md.Method == http.MethodGet)
 			merr = append(merr, errs...)
-			_, _, errs = validateIngressRequestOrResponse(scopes, module, n, "response", n.Response)
+			errs = validateIngressResponse(scopes, module, n, "response", n.Response)
 			merr = append(merr, errs...)
 
-			// Validate path
-			for _, path := range md.Path {
-				switch path := path.(type) {
-				case *IngressPathParameter:
-					reqBodyData, ok := reqBody.(*Data)
-					if !ok {
-						merr = append(merr, errorf(path, "ingress verb %s: cannot use path parameter %q with request type %s, expected Data type", n.Name, path.Name, reqBodyType))
-					} else if reqBodyData.FieldByName(path.Name) == nil {
-						merr = append(merr, errorf(path, "ingress verb %s: request type %s does not contain a field corresponding to the parameter %q", n.Name, reqBodyType, path.Name))
+			if reqInfo.pathParamSymbol != nil {
+				// If this is nil it has already failed validation
+
+				hasParameters := false
+				// Validate path
+				for _, path := range md.Path {
+					switch path := path.(type) {
+					case *IngressPathParameter:
+						hasParameters = true
+						switch dataType := reqInfo.pathParamSymbol.(type) {
+						case *Data:
+							if dataType.FieldByName(path.Name) == nil {
+								merr = append(merr, errorf(path, "ingress verb %s: request pathParameter type %s does not contain a field corresponding to the parameter %q", n.Name, reqInfo.pathParamType, path.Name))
+							}
+						case *Map:
+						// No validation for map, it is always fine
```

Comment:
We should probably validate the key type is string for query and path parameters.